qutip/solver/result.py

```diff
@@ -71,41 +73,43 @@ def _read_e_ops(self, _super):
             e_ops = [e for e in self._raw_e_ops.values()]
         elif callable(self._raw_e_ops):
             e_ops = [self._raw_e_ops]
+        elif self._raw_e_ops is None:
```

Comment:
Ah, sorry, you are right. What I did for the BoFiN PR was to always keep the results in the same form as e_ops, rather than converting the results back to a dict at the end.

Since this isn't a user facing change, happy to leave it our for now.